pkg/autoscaler/multiscaler.go

```diff
@@ -195,12 +196,27 @@ func (m *MultiScaler) Delete(ctx context.Context, namespace, name string) error
 
 // Watch registers a singleton function to call when MetricStatus is updated.
 func (m *MultiScaler) Watch(fn func(string)) {
+	m.watcherMutex.Lock()
+	defer m.watcherMutex.Unlock()
+
 	if m.watcher != nil {
 		m.logger.Fatal("Multiple calls to Watch() not supported")
 	}
 	m.watcher = fn
 }
 
+// Inform sends an update to the registered watcher function, if it is set.
+func (m *MultiScaler) Inform(event string) bool {
+	m.watcherMutex.RLock()
+	defer m.watcherMutex.RUnlock()
+
+	if m.watcher != nil {
+		m.watcher(event)
+		return true
+	}
+	return false
```

Comment:
Maybe better to return an error and log it for posterity? it should never happen but just in case